### PR TITLE
[LUPEYALPHA-652] FE ineligible provider selected

### PR DESCRIPTION
**Pull request by asmega**
# Context

- https://dfedigital.atlassian.net/browse/LUPEYALPHA-652
- Also bug https://dfedigital.atlassian.net/browse/LUPEYALPHA-664
- When a user selects an FE provider that is not eligible show them ineligible state
- Handle when closed FE provider is selected

# Changes

- When user does provider search we now store this as `possible_school_id` then upgrade to `school_id` when they confirm it on the second screen. This is needed otherwise will show ineligible page at wrong time if they go back and change to a different FE provider
- Tested both JS and non-JS journeys. If doing the non-JS journey we assume that if they submit their search they are always changing their answer therefore will always have to confirm their school even if it is the same FE provider
- Closed providers are now returnable via the search however if the user selects such a provider they see the ineligible screen that has now been implemented
- Updated seeding of review apps so users can test FE journey on review apps, otherwise they would always see ineligible screen

### CODE REVIEW

**Review by slorek — COMMENTED** [review]
How does `possible_school_id` work when there can be more than one matching school to choose from?

**Comment by asmega** [comment]
> How does `possible_school_id` work when there can be more than one matching school to choose from?

this is used for the JS version, if you click an option from the autocomplete dropdown it'll set `possible_school_id` to the clicked option

File: app/models/journeys/further_education_payments/session_answers.rb [hunk 1]
```diff
@@ -66,6 +75,27 @@ def lacks_teacher_qualification_or_enrolment?
       def less_than_half_hours_teaching_fe?
         half_teaching_hours == false
       end
+
+      def eligible_fe_provider?
+        return unless school
```
[the remark this answers is not in the document]
Comment (asmega) [reply app/models/journeys/further_education_payments/session_answers.rb:84]: i think anything falsey should be fine here

File: app/models/journeys/further_education_payments/session_answers.rb [hunk 2]
```diff
@@ -66,6 +75,27 @@ def lacks_teacher_qualification_or_enrolment?
...
+
+        EligibleFeProvider
+          .where(academic_year: AcademicYear.current)
+          .where(ukprn: school.ukprn)
+          .exists?
+      end
+
+      def ineligible_fe_provider?
+        return unless school
```
[the remark this answers is not in the document]
Comment (asmega) [reply app/models/journeys/further_education_payments/session_answers.rb:93]: i think anything falsey should be fine here

File: app/models/journeys/further_education_payments/session_answers.rb [hunk 3]
```diff
@@ -66,6 +75,27 @@ def lacks_teacher_qualification_or_enrolment?
...
+
+        !eligible_fe_provider?
+      end
+
+      def fe_provider_closed?
+        return unless school
```
[the remark this answers is not in the document]
Comment (asmega) [reply app/models/journeys/further_education_payments/session_answers.rb:99]: i think anything falsey should be fine here

**Review by slorek — APPROVED** [review]

**Pull request merged**